R/compat-tidyr.R

```diff
@@ -0,0 +1,9 @@
+strip_names <- function(df, base, names_sep) {
+  base <- paste0(base, names_sep)
+  names <- names(df)
+
+  has_prefix <- regexpr(base, names, fixed = TRUE) == 1L
```

Comment:
Took me a second to parse this — maybe `substr(names, 1, nchar(base)) == base` is simpler?

Reply:
I just remembered there is the much simpler `startsWith(names, base)`.

Reply:
You can't trust past-Hadley 😉  Also cool that we can finally use `startsWith()` — I think it was added in 3.3.